packages/sdk/src/Confidence.ts

```diff
@@ -73,17 +93,26 @@ export class Confidence implements EventSender, Trackable {
   }
 
   setContext(context: Context): void {
+    const current = this.getContext();
+    const changedKeys: string[] = [];
     for (const key of Object.keys(context)) {
-      this.updateContextEntry(key, context[key]);
+      if (Value.equal(current[key], context[key])) continue;
+      changedKeys.push(key);
+      this._context.set(key, Value.clone(context[key]));
+    }
+    if (this.contextChanged && changedKeys.length > 0) {
+      this.contextChanged(changedKeys);
     }
-  }
-
-  private updateContextEntry<K extends string>(name: K, value: Context[K]) {
-    this._context.set(name, Value.clone(value));
   }
 
   clearContext(): void {
```

Comment:
We should remove this function. It's never really useful and it complicates things.

Reply:
Maybe later...